dash/include/dash/algorithm/ForEach.h

```diff
@@ -28,17 +28,14 @@ namespace dash {
  *
  * \ingroup     DashAlgorithms
  */
-template <
-  typename ElementType,
-  class    PatternType,
-  class    UnaryFunction >
+template <typename GlobInputIt, class UnaryFunction>
 void for_each(
-  /// Iterator to the initial position in the sequence
-  const GlobIter<ElementType, PatternType> & first,
-  /// Iterator to the final position in the sequence
-  const GlobIter<ElementType, PatternType> & last,
-  /// Function to invoke on every index in the range
-  UnaryFunction                              func)
+    /// Iterator to the initial position in the sequence
+    const GlobInputIt& first,
+    /// Iterator to the final position in the sequence
+    const GlobInputIt& last,
+    /// Function to invoke on every index in the range
+    UnaryFunction func)
 {
   /// Global iterators to local index range:
   auto index_range  = dash::local_index_range(first, last);
```

Comment:
Can we statically check whether `GlobInputIt` is a global pointer to avoid ugly compiler errors? (I think it's done in `generate` already)

Reply:
Done